TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,646 @@
+    public DriveTrain(DcMotor leftFront, DcMotor rightFront, DcMotor leftRear, DcMotor rightRear, AHRS gyro, Telemetry telemetry) {
+        telemetry.addData("DriveTrain Startup", "Beginning");
+        telemetry.update();
+        //setup for all the motors.
+        //setup for all the front motors.
+        this.leftFront = leftFront;
+        //setup for the left front motor.
+        this.rightFront = rightFront;
+        //setup for the right front motor.
+        //setup for all the back motors.
+        this.leftRear = leftRear;
+        //setup for the left back motor.
+        this.rightRear = rightRear;
+        //setup for the right back motor.
+
+        leftFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        leftRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        //sets all the motors power to zero, this prevents the motors from spinning at the beginning of the program
+        this.leftFront.setPower(0);
+        this.rightFront.setPower(0);
+        this.leftRear.setPower(0);
+        this.rightRear.setPower(0);
+
+        //gyro sensor setup.
+        //this.gyro = AHRS.getInstance(hardwareMap.deviceInterfaceModule.get("dim"), gyroPort, AHRS.DeviceDataType.kProcessedData);
+        this.gyro = gyro;
+        //sets up the gyro sensor.
+        this.gyro.zeroYaw();
+        gyroReset = gyro.getDimStateTrackerInstance();
+
+        //sets the gyro sensors position to zero.
+        gyroReset.reset();
+
+        //misc setup
+
+        this.goalDegrees = -1;
+        this.goalPosition = -1;
+        this.gyroRange = 3;
+
+        this.telemetry = telemetry;
+        this.encodersCanRun = true;
+        this.goalEncoderPosition = -1;
+        goalBackwardPosition = -1;
+        telemetry.addData("DriveTrain Startup", "End");
+        telemetry.update();
+    }
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    //MOVE
+    public void setMotorPower(double x, double y, double z){
+        /*
+        Guide to motor Powers:
+        Left Front: - (y + x + Z)
+        Right Front: y - x - z
+        Left Rear: y + x - z
+        Right Rear: - (y - x + z)
+         */
+        this.leftFront.setPower(Range.clip(-(y+x+z),-1,1));
+        this.rightFront.setPower(Range.clip((y-x-z),-1,1));
+        this.leftRear.setPower(Range.clip(-(y-x+z),-1,1));
+        this.rightRear.setPower(Range.clip((y+x-z), -1, 1));
+    }
+
+    public void zonedDrive(int zone, double x, double y, double z) {
+        power = Math.sqrt((x*x) + (y*y));
+        switch(zone) {
+            case 0:
```

Comment:
#add a todo to come back and refactor this code later to get rid of these magic case numbers.